enhancements/api-review/azure_user_defined_tags.md

```diff
@@ -0,0 +1,322 @@
+                              maxLength: 128
+                              minLength: 1
+                              pattern: ^[a-zA-Z][0-9A-Za-z_.=+-@]+$
+                            value:
+                              description: value is the value of the tag.
+                              type: string
+                              maxLength: 256
+                              minLength: 1
+                              pattern: ^[0-9A-Za-z_.=+-@]+$
+```
+
+### Implementation Details/Notes/Constraints [optional]
+Add a new field `resourceTags` to `.status.platformStatus.azure` of the 
+`infrastructure.config.openshift.io` type. Tags included in the `resourceTags` field 
+will be applied to new resources created for the cluster by the in-cluster operators.
+
+The `resourceTags` field in `status.platformStatus.azure` of `infrastructure.config.openshift.io`
+will be populated by the installer using the entries from `platform.azure.userTags` field of `install-config`.
+
+`status.platformStatus.azure` of `infrastructure.config.openshift.io` is immutable.
+
+All operators that create Azure resources will apply these tags to all Azure 
+resources they create.
+
+| Operator | Resources created by the operator |
+| -------- | ----------------------------- |
+| cloud-network-config-controller | Private IP address |
+| cluster-image-registry-operator | Storage Account |
+| cluster-ingress-operator | Load Balancer, DNS records |
+| cloud-credential-operator | IAM roles and policies |
+| machine-api-provider-azure | Application Security Group, Availability Set, Group, Load Balancer, Public IP Address, Route, Network Security Group, Virtual Machine Extension, Virtual Interface, Virtual Machine, Virtual Network. |
+
+Below list of terraform Azure APIs to create resources should be updated to add user
+defined tags and as well the openshift default tag in the installer component.
+`azurerm_resource_group, azurerm_image, azurerm_lb, azurerm_network_security_group, azurerm_storage_account, azurerm_user_assigned_identity, azurerm_virtual_network, azurerm_linux_virtual_machine, azurerm_network_interface, azurerm_dns_cname_record`
+
+API update example:
+A local variable should be defined, which merges the default tag and the user
+defined Azure tags, which should be referred in the Azure resource APIs.
+``` terraform
+locals {
+  tags = merge(
+    {
+      "kubernetes.io_cluster.${var.cluster_id}" = "owned"
+    },
+    var.azure_extra_tags,
+  )
+}
+
+resource "azurerm_resource_group" "main" {
+  tags     = local.tags
+}
+```
+
+#### Caveats
+1. User updates or removes Azure resource tags added by Openshift, using an external interface
+   will not be reconciled by the resource owning operator.
+2. Updating tags of individual resources is not supported and any tag present in 
+   `.status.platformStatus.azure.resourceTags` of `infrastructure.config.openshift.io/v1` resource 
+   will result in addding tags to all Openshift managed Azure resources. 
```

Comment:
```suggestion
   will result in adding tags to all Openshift managed Azure resources. 
```

Reply:
also not sure this statement is true.  If the resource has been created, and then someone updates spec.usertags, status.usertags will reflect the spec update(right?), but already created resources may or may not have the new tag added (depending on what the component that manages the resource decides to do, and also whether or not it's actually possible to update the tags on an existing azure resource)

Reply:
What I wanted to suggest was, defining tags for individual resources is not supported, but all the user defined tags will be applied to all the resources that will be created. I will rephrase the statement.